lib/api/2.0/users.js

```diff
@@ -49,45 +48,48 @@ var addUser = controller({success: 201}, function(req, res) {
 });
 
 var modifyUser = controller(function(req) {
-    var userObj = _.pick(req.swagger.params.body.value, ['username', 'password', 'role']);
-    if(!req.isAuthenticated) {
-        throw new Errors.UnauthorizedError('Unauthorized');
-    }
+    var name = req.swagger.params.name.value;
+    var userObj = _.pick(req.swagger.params.body.value, ['password', 'role']);
 
-    if(!req.isAuthenticated()) {
-        throw new Errors.UnauthorizedError('Unauthorized');
-    }
+    return Promise.try(function() {
+        if(!req.isAuthenticated || !req.isAuthenticated()) {
```

Comment:
What if `isAuthenticated` is falsey but not a function? Won't we throw a `TypeError` here rather than the intended `UnauthorizedError`?